### PR TITLE
CWE Top 25 2019 & 2020

### PR DESCRIPTION
**Pull request by shaopeng-gh**

### CODE REVIEW

File: Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs [hunk 1]
```diff
@@ -139,10 +139,10 @@ private Run ConvertToSarif(List<CweCsvRecord> records, string version)
...
             {
-                Name = Constants.CWE.Name,
-                Guid = Constants.CWE.Guid,
+                Name = Constants.CWE_Comprehensive_V43.Name,
+                Guid = Constants.CWE_Comprehensive_V43.Guid,
                 Version = version,
-                ReleaseDateUtc = Constants.CWE.ReleaseDate,
+                ReleaseDateUtc = Constants.CWE_Comprehensive_V43.ReleaseDate,
                 InformationUri = new Uri("https://cwe.mitre.org/data/published/cwe_v4.4.pdf"),
                 DownloadUri = new Uri("https://cwe.mitre.org/data/xml/cwec_v4.4.xml.zip"),
```
Comment (eddynaka) [inline Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs]: >DownloadUri [](http://example.com/codeflow?start=16&length=11)

this does not look right.
The version is 43, but its point to 44 #Closed

Comment (shaopeng-gh) [reply Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs]: forgot to change the url, fixed

**Comment by eddynaka** [comment]
>         private Run ConvertToSarif(List<CweCsvRecord> records, string version)

should we delete this? i dont see any usage for the csv method

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293)

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293)

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293)

---
Refers to: Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs:137 in f6fc3f5. [](commit_id = f6fc3f579ab923125d350149a552ce2274edcb40, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>         private Run ConvertToSarif(List<CweCsvRecord> records, string version)

we have csv code and xml code, currently csv code is not used and I think it is best to remove it, and if we need it we can still get the code form history.
I have removed it.
also I see there are 3 wanrings in the code, I used VS suggested fix and fixed them.

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293,868445293)

---
Refers to: Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs:137 in f6fc3f5. [](commit_id = f6fc3f579ab923125d350149a552ce2274edcb40, deletion_comment = False)

**Review by eddynaka — APPROVED** [review]
:shipit:

**Pull request merged**